utils/test-transform.js

```diff
@@ -0,0 +1,78 @@
+const babel = require("babel-core");
+
+const unpad = require("./unpad");
+
+function _transform(source, options) {
+  return babel.transform(unpad(source), options).code.trim();
+}
+
+function makeTester(
+  plugins,
+  opts,
+  { transform = _transform, check },
+  excludeKeys = []
+) {
+  if (!Array.isArray(plugins)) {
+    plugins = [plugins];
+  }
+  const thePlugin = (name, source, expected = source, babelOpts) => {
+    if (typeof expected === "object") {
+      babelOpts = expected;
+      expected = source;
+    }
+    const { stack } = new Error();
+    const options = Object.assign(
+      { plugins, sourceType: "script" },
+      opts,
+      babelOpts
+    );
+    it(name, () => {
+      const transformed = transform(source, options);
+      try {
+        check({
+          transformed,
+          expected: unpad(expected),
+          source: unpad(source)
+        });
+      } catch (e) {
+        // use the stack from outside the it() clause
+        // (the one inside the clause doesn’t show the actual test code)
+        e.stack = stack;
+        throw e;
+      }
+    });
+  };
+  thePlugin.skip = name => it.skip(name);
```

Comment:
you would also need to support `only` if we would like to run a single test during dev..